### ISSUE TITLE
Raw option list gets fully expanded

### ISSUE THREAD

**Issue by Skillmon** [latex3/latex2e#617]
## Brief outline of the bug

The raw option list currently gets fully expanded inside an `\edef` assignment once, instead of containing the unprocessed options.

## Minimal example showing the bug

```latex
\RequirePackage{latexbug} 
\begin{filecontents}{mytmp.sty}
\ProvidesPackage{mytmp}
\expandafter\show\csname @raw@opt@\@currname.\@currext\endcsname
\stop
\end{filecontents}
\def\emptyTWO{\empty}
\RequirePackage[one=\emptyTWO,two=\unexpanded{\emptyTWO}]{mytmp}\relax
```

## Log file (required) and possibly PDF file  

- [Log file](https://github.com/latex3/latex2e/files/6804091/rawopstest.log)

**Comment by davidcarlisle** [comment]
I think this is clearly a bug, the raw list is not supposed to be expanded. 

I think it is enough to add `\unexpanded` as in

```
\begin{filecontents}{mytmp.sty}
\ProvidesPackage{mytmp}
\expandafter\show\csname opt@\@currname.\@currext\endcsname
\expandafter\show\csname @raw@opt@\@currname.\@currext\endcsname
\end{filecontents}

\makeatletter

\def\@fileswith@pti@ns#1[#2]#3[#4]{%
  \ifx#1\@clsextension
    \ifx\@classoptionslist\relax
      \xdef\@classoptionslist{\zap@space#2 \@empty}%
      \gdef\@raw@classoptionslist{#2}%
      \def\reserved@a{%
        \@onefilewithoptions#3[{#2}][{#4}]#1%
        \@documentclasshook}%
    \else
      \def\reserved@a{%
        \@onefilewithoptions#3[{#2}][{#4}]#1}%
    \fi
  \else
    \def\reserved@b##1,{%
      \ifx\@nnil##1\relax\else
        \ifx\@nnil##1\@nnil\else
         \noexpand\@onefilewithoptions##1[{\unexpanded{#2}}][{#4}]%
                                          %%%%%%%%%%%%
         \noexpand\@pkgextension
        \fi
        \expandafter\reserved@b
      \fi}%
      \edef\reserved@a{\zap@space#3 \@empty}%
      \edef\reserved@a{\expandafter\reserved@b\reserved@a,\@nnil,}%
  \fi
  \reserved@a}


\def\emptyTWO{\empty}
\RequirePackage[one=\emptyTWO,two=\unexpanded{\emptyTWO, a5 paper}]{mytmp}\relax



\stop
```

**Comment by davidcarlisle** [comment]
fixed by PR #620
